### ISSUE TITLE
[release/2.13] timm_models_accuracy_training_amp-fp16 fbnetv3_b fail_accuracy

### ISSUE THREAD

**Issue by bjarzemb** [intel/torch-xpu-ops#4018]
### 🐛 Describe the bug

There is fail_accuracy on Windows with Intel XPU (Arc B580) for timm_models_accuracy_training_amp-fp16 fbnetv3_b.

Output:
```
loading model: 0it [00:00, ?it/s]
loading model: 0it [00:05, ?it/s]
xpu  train fbnetv3_b                          
E0615 07:30:26.668000 2836 site-packages\torch\_dynamo\utils.py:3706] RMSE (res-fp64): 0.00472, (ref-fp64): 0.00073 and shape=torch.Size([16]). res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000, use_larger_multiplier_for_smaller_tensor: 0
E0615 07:30:26.669000 2836 site-packages\torch\_dynamo\utils.py:3538] Accuracy failed for key name blocks.0.0.bn1.weight.grad
fail_accuracy
```

### Versions

<details>
<summary>Click to expand environment details</summary>

```
PyTorch version: 2.13.0+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro (10.0.26100 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: N/A

Python version: 3.12.13 | packaged by conda-forge | (main, Mar  5 2026, 16:36:12) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26100-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260000
Intel GPU driver version:
* 32.0.101.8826 (20260529000000.******+***)
Intel GPU models onboard:
* Intel(R) Arc(TM) B580 Graphics
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Arc(TM) B580 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE20B, uuid=86800be2-0000-0000-0300-000000000000, driver_version='1.15.37858', total_memory=11875MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=0MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Name: 13th Gen Intel(R) Core(TM) i5-13400
Manufacturer: GenuineIntel
Family: 205
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 2500
MaxClockSpeed: 2500
L2CacheSize: 9728
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] dpcpp-cpp-rt==2026.0.0
[pip3] intel-cmplr-lib-rt==2026.0.0
[pip3] intel-cmplr-lib-ur==2026.0.0
[pip3] intel-cmplr-lic-rt==2026.0.0
[pip3] intel-opencl-rt==2026.0.0
[pip3] intel-openmp==2026.0.0
[pip3] intel-pti==0.17.0
[pip3] intel-sycl-rt==2026.0.0
[pip3] mkl==2026.0.0
[pip3] mkl-include==2024.2.0
[pip3] mkl-static==2024.2.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.2
[pip3] onemkl-license==2026.0.0
[pip3] onemkl-sycl-blas==2026.0.0
[pip3] onemkl-sycl-dft==2026.0.0
[pip3] onemkl-sycl-lapack==2026.0.0
[pip3] onemkl-sycl-rng==2026.0.0
[pip3] onemkl-sycl-sparse==2026.0.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2023.0.0
[pip3] tbb-devel==2021.13.1
[pip3] tcmlib==1.5.0
[pip3] torch==2.13.0+xpu
[pip3] torchaudio==2.11.0+xpu
[pip3] torchmetrics==1.9.0
[pip3] torchrec-nightly==2022.4.26
[pip3] torchvision==0.28.0+xpu
[pip3] torchviz==0.0.3
[pip3] torchx-nightly==2026.6.11
[pip3] triton-xpu==3.7.2
[pip3] umf==1.1.0
[conda] dpcpp-cpp-rt              2026.0.0                 pypi_0    pypi
[conda] intel-cmplr-lib-rt        2026.0.0                 pypi_0    pypi
[conda] intel-cmplr-lib-ur        2026.0.0                 pypi_0    pypi
[conda] intel-cmplr-lic-rt        2026.0.0                 pypi_0    pypi
[conda] intel-opencl-rt           2026.0.0                 pypi_0    pypi
[conda] intel-openmp              2026.0.0                 pypi_0    pypi
[conda] intel-pti                 0.17.0                   pypi_0    pypi
[conda] intel-sycl-rt             2026.0.0                 pypi_0    pypi
[conda] mkl                       2026.0.0                 pypi_0    pypi
[conda] mkl-include               2024.2.0                 pypi_0    pypi
[conda] mkl-static                2024.2.0                 pypi_0    pypi
[conda] numpy                     1.26.2                   pypi_0    pypi
[conda] onemkl-license            2026.0.0                 pypi_0    pypi
[conda] onemkl-sycl-blas          2026.0.0                 pypi_0    pypi
[conda] onemkl-sycl-dft           2026.0.0                 pypi_0    pypi
[conda] onemkl-sycl-lapack        2026.0.0                 pypi_0    pypi
[conda] onemkl-sycl-rng           2026.0.0                 pypi_0    pypi
[conda] onemkl-sycl-sparse        2026.0.0                 pypi_0    pypi
[conda] optree                    0.13.0                   pypi_0    pypi
[conda] tbb                       2023.0.0                 pypi_0    pypi
[conda] tbb-devel                 2021.13.1                pypi_0    pypi
[conda] tcmlib                    1.5.0                    pypi_0    pypi
[conda] torch                     2.13.0+xpu               pypi_0    pypi
[conda] torchaudio                2.11.0+xpu               pypi_0    pypi
[conda] torchmetrics              1.9.0                    pypi_0    pypi
[conda] torchrec-nightly          2022.4.26                pypi_0    pypi
[conda] torchvision               0.28.0+xpu               pypi_0    pypi
[conda] torchviz                  0.0.3                    pypi_0    pypi
[conda] torchx-nightly            2026.6.11                pypi_0    pypi
[conda] triton-xpu                3.7.2                    pypi_0    pypi
[conda] umf                       1.1.0                    pypi_0    pypi
```

</details>

**Comment by mengfei25** [comment]
On B60 Linux also fail_accuracy https://github.com/intel/torch-xpu-ops/issues/4001#issuecomment-4714089578

**Comment by chuanqi129** [comment]
@jianyizh use this dedicated issue to track the new accuracy failure on BMG linux and Windows, please aware it

**Comment by bjarzemb** [comment]
I have this issue also on WSL2 with Intel XPU (Arc B580) in release 2.13 tests.

**Comment by weishi-deng** [comment]
cause: https://github.com/pytorch/pytorch/pull/184366 This prevents inductor from removing `fp32→fp16→fp32` cast chains. Preserving fp16 intermediate values causes precision loss in BatchNorm and  https://github.com/pytorch/pytorch/pull/184110, which config (enabled by default since dev20260603) causes forward intermediate results to be additionally truncated to fp16. Both reverts can pass on latest nightly.

**Comment by weishi-deng** [comment]
cuda also fail(both before these changes and after)

**Comment by weishi-deng** [comment]
issue: 188261

**Comment by chuanqi129** [comment]
@weishi-deng, I noticed that there is a draft PR 188992 for issue 188261, could you please check it?

**Comment by weishi-deng** [comment]
No response from 188992. Also, PR 188992 just fixed part of the issue: fp32→fp16→fp32 cast chains. I commented on the PR to ask for a fix of 184110.
